### PR TITLE
DM-5382: Add Python, C++ and Plain Text stack preambles, plus COPYRIGHT and LICENSE templates

### PR DESCRIPTION
**Pull request by jonathansick**
These preambles are based on [RFC-45](https://jira.lsstcorp.org/browse/RFC-45).

- New `templatekit.textutils.reformat_content_lines` to apply a Python format expression to each line of string.
- New `line_format_builder` Scons builder that can be used to generate the C++ and Python preambles from the plain text preambles by formatting with text with `reformat_content_lines`.
- Add new `stack_license_preamble_cpp`, `stack_license_preamble_py`, and `stack_license_preamble_txt` file templates (the C++ and Python templates are Scons-generated from the txt version)
- Remove the old-style license boilerplate templates.
- Add template for COPYRIGHT files.
- Add template for the pure GPLv3 LICENSE file.

### CODE REVIEW

File: file_templates/stack_license_preamble_py/template.py.jinja [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
-# 
-# You should have received a copy of the LSST License Statement and 
-# the GNU General Public License along with this program.  If not, 
-# see <https://www.lsstcorp.org/LegalNotices/>.
 #
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
```
Comment (ktlim) [inline file_templates/stack_license_preamble_py/template.py.jinja]: And here (or I guess in the code that generates these).

**Comment by jonathansick** [comment]
Yep, they are derived from the txt preamble. See https://github.com/lsst/templates/pull/7/files#diff-22a68eea63491b97bda4202db7ba2eabR7

When the txt preamble changes, it triggers a change in the py and cc preambles because of the dependency established by scons. Travis runs scons and ensures that all committed templates and examples are consistent.

**Comment by timj** [comment]
Ok. I couldn't see travis files because I hadn't realized this was a PR against #6 and not against master. The travis file does run scons but is testing that nothing changed. Does this mean that the idea is that scons generates the files and the person changing the templates commits the generated files? Then travis tests that things are synced up?

**Comment by jonathansick** [comment]
Exactly. These are relevant docs for doing that:

- https://github.com/lsst/templates/blob/tickets/DM-5382/CONTRIBUTING.md#set-up-for-development
-  https://github.com/lsst/templates/blob/tickets/DM-5382/CONTRIBUTING.md#template-example-file
- https://github.com/lsst/templates/blob/tickets/DM-5382/CONTRIBUTING.md#project-template-example-directory
- https://github.com/lsst/templates/blob/tickets/DM-5382/CONTRIBUTING.md#continuous-integration

**Pull request merged**